torchtune/models/clip/_position_embeddings.py

```diff
@@ -137,6 +138,10 @@ def _load_state_dict_hook(
         inpt_local_pos_embed = state_dict.get(
             prefix + "local_token_positional_embedding"
         )
+        local_device = inpt_local_pos_embed.device
+        if isinstance(inpt_local_pos_embed, DTensor):
+            inpt_local_pos_embed = inpt_local_pos_embed.full_tensor()
+
         if inpt_local_pos_embed is not None:
```

Comment:
we should probably add here: 

if inpt_local_pos_embed is not None and inpt_local_pos_embed.shape != self. local_token_positional_embedding

But testing becomes a bit trickier. Maybe for now its better to not add it until testing is completed

Reply:
Yeah will leave out for now

Reply:
i think we should add before the PR is finalized though. What do you think?

Reply:
Why do you say testing becomes trickier?

Reply:
After thinking about it more, I don't think we should add it. The DTensor fix resolves the issue and no need to add extra logic on top of this that was not present before